Framework/Algorithms/src/ConjoinWorkspaces.cpp

```diff
@@ -145,6 +133,57 @@ void ConjoinWorkspaces::checkForOverlap(const MatrixWorkspace &ws1,
   }
 }
 
+/**
+ * Conjoin two event workspaces together, including the history
+ *
+ * @param ws1:: The first workspace
+ * @param ws2:: The second workspace
+ * @return :: A new workspace containing the conjoined workspaces
+*/
+API::MatrixWorkspace_sptr
+ConjoinWorkspaces::conjoinEvents(const DataObjects::EventWorkspace &ws1,
+                                 const DataObjects::EventWorkspace &ws2) {
+  this->validateInputs(ws1, ws2, false);
+
+  // Check there is no overlap
+  if (this->getProperty("CheckOverlapping")) {
+    this->checkForOverlap(ws1, ws2, false);
+    m_overlapChecked = true;
```

Comment:
Do we need `m_overlapChecked` it is only used in `fixSpectrumNumbers():202` but don't all paths lead to this being checked anyway?